frontend/src/store/modules/shoots/getters.js

```diff
@@ -138,6 +143,52 @@ export function getSortVal (rootGetters, item, sortBy) {
   }
 }
 
+export const conditionCache = {
```

Comment:
I still think this is not clean because we access this data in a getter and we still need this functional getter. I would propose to move/add the following code to `helper.js`.
```js
export const knownConditions = {
  APIServerAvailable: Object.freeze({
    name: 'API Server',
    shortName: 'API',
    description: 'Indicates whether the shoot\'s kube-apiserver is healthy and available. If this is in error state then no interaction with the cluster is possible. The workload running on the cluster is most likely not affected.',
    weight: '0'
  }),
  ControlPlaneHealthy: Object.freeze({
    name: 'Control Plane',
    shortName: 'CP',
    description: 'Indicates whether all control plane components are up and running.',
    showAdminOnly: true,
    weight: '1'
  }),
  EveryNodeReady: Object.freeze({
    name: 'Nodes',
    shortName: 'N',
    description: 'Indicates whether all nodes registered to the cluster are healthy and up-to-date. If this is in error state there then there is probably an issue with the cluster nodes. In worst case there is currently not enough capacity to schedule all the workloads/pods running in the cluster and that might cause a service disruption of your applications.',
    weight: '3'
  }),
  SystemComponentsHealthy: Object.freeze({
    name: 'System Components',
    shortName: 'SC',
    description: 'Indicates whether all system components in the kube-system namespace are up and running. Gardener manages these system components and should automatically take care that the components become healthy again.',
    weight: '2'
  }),
  ObservabilityComponentsHealthy: Object.freeze({
    name: 'Observability Components',
    shortName: 'OC',
    description: 'Indicates whether all observability components like Prometheus, Loki, Grafana, etc. are up and running. Gardener manages these system components and should automatically take care that the components become healthy again.',
    weight: '4'
  }),
  MaintenancePreconditionsSatisfied: Object.freeze({
    name: 'Maintenance Preconditions Satisfied',
    shortName: 'M',
    description: 'Indicates whether Gardener is able to perform required actions during maintenance. If you do not resolve this issue your cluster will eventually turn into an error state.',
    weight: '5'
  }),
  HibernationPossible: Object.freeze({
    name: 'Hibernation Preconditions Satisfied',
    shortName: 'H',
    description: 'Indicates whether Gardener is able to hibernate this cluster. If you do not resolve this issue your hibernation schedule may not have any effect.',
    weight: '6'
  })
}

function createCondition (type) {
  const conditionComponents = snakeCase(type).split('_')
  const dropSuffixes = [
    'available',
    'healthy',
    'ready',
    'availability'
  ]
  if (dropSuffixes.includes(conditionComponents[conditionComponents.length - 1])) {
    conditionComponents.pop()
  }
  const words = []
  const letters = []
  for (const conditionComponent of conditionComponents) {
    const word = upperFirst(conditionComponent)
    words.push(word)
    letters.push(word[0])
  }
  const name = words.join(' ')
  const shortName = letters.join('')
  return Object.freeze({
    name,
    shortName,
    weight: shortName
  })
}

export function setConditionTypes (state, object) {
  const setConditionType = ({ type }) => {
    if (!state.conditions[type]) {
      Vue.set(state.conditions, type, createCondition(type))
    }
  }
  object.status
    ?.conditions
    ?.forEach(setConditionType)
  object.status
    ?.constraints
    ?.filter(({ codes }) => codes?.length)
    .forEach(setConditionType)
}
```
We add the list of conditions to the shoot state and whenever we update the shoots in the store, we update the list of conditions. Additionally we add a new mutation SET_CONDITION_TYPES which could be used in the test to update the store before shallow mounting a component.
```
  RECEIVE (state, { rootState, rootGetters, shoots: items }) {
     ...
    state.shoots = shoots
    for (const object of Object.values(shoots)) {
      setConditionTypes(state, object)
    }
    ...
  },
  ...
  HANDLE_EVENT (state, { rootState, rootGetters, event }) {
          ...
          putItem(state, event.object)
          setConditionTypes(state, event.object)
  },
  ...
  SET_CONDITION_TYPES (state, object) {
    setConditionTypes(state, object)
  }
```
We access the conditions object via the shoots module state in the sorting 
```js
     const condition = state.conditions[itemCondition.type]
```
and in the StatusTags.
```js
      return sortBy(this.shootReadiness
        .filter(condition => !!condition.lastTransitionTime)
        .map(condition => ({
          ...this.$store.state.shoots.conditions[condition.type],
          ...condition
        })), 'weight')
```
In the test `store.shoots.spec.js` we initialize the store conditions in the `beforeEach` function.
```js
  beforeEach(() => {
    ...
    state = {
      ...
      conditions: { ...knownConditions }
    }
    for (const object of shootItems) {
      setConditionTypes(state, object)
    }
    ...
  })  
```
In the test `StatusTags.spec.js` we do it in `shallowMountStatusTags` and in `beforeEach`.
```js
  const shallowMountStatusTags = conditionTypes => {
    const conditions = conditionTypes.map(type => ({ type }))
    store.commit('shoots/SET_CONDITION_TYPES', { status: { conditions } })
    ...
  }
  
  beforeEach(() => {
    vuetify = new Vuetify()
    const state = cloneDeep(shootModule.state)
    Object.assign(state.conditions, {
      ControlPlaneHealthy: {
        name: 'Control Plane Overwritten',
        shortName: 'CPO',
        description: 'Overwritten Description'
      },
      ConditionFromConfigAvailability: {
        name: 'Config Condition',
        shortName: 'CC',
        description: 'Config Condition Description'
      },
      ImportantCondition: {
        name: 'Important Condition',
        shortName: 'IC',
        description: 'Important Config Condition Description',
        weight: '0'
      }
    })
    store = new Vuex.Store({
      state: {},
      modules: {
        shoots: {
          ...shootModule,
          state
        }
      }
    })
  })
```
What do you think?